catroid/src/main/java/org/catrobat/catroid/ui/fragment/FragmentFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.ui.fragment;
+
+import android.app.Fragment;
+import android.content.Context;
+
+final class FragmentFactory {
+
+	private FragmentFactory() {
+	}
+
+	static FormulaEditorCategoryListFragment createFormulaEditorCategoryListFragment(Context context) {
+		String fragmentName = FormulaEditorCategoryListFragment.flavoredTag;
+		return (FormulaEditorCategoryListFragment) FragmentFactory.createFragment(fragmentName, context);
+	}
+
+	static BrickCategoryFragment createBrickCategoryFragment(Context context) {
+		String fragmentName = BrickCategoryFragment.flavoredTag;
+		return (BrickCategoryFragment) FragmentFactory.createFragment(fragmentName, context);
+	}
+
+	private static Fragment createFragment(String className, Context context) {
+		String classNameWithPackage = context.getPackageName() + "." + className;
```

Comment:
i dont know a lot about the whole flavoring thing, but is this really supposed to be done like this? (like building the full class and package as string and the loading it that way)